soh/soh/Enhancements/randomizer/3drando/hints.cpp

```diff
@@ -998,113 +935,76 @@ static void DistrabuteHints(std::array<uint8_t, HINT_TYPE_MAX>& selected, uint8_
+        if (stoneCount >= distTable[distribution].copies){
+          selected[distribution] += 1;
+          stoneCount -= distTable[distribution].copies;
           break;
         }
         else {
-          totalWeight -= distTable[hintType].weight;
-          distTable[hintType].weight = 0;
+          totalWeight -= distTable[distribution].weight;
+          distTable[distribution].weight = 0;
           break;
         }
       }
     }
+    //if there's still weight, then it's junk
+    if (currentWeight > 0){
+      selected[selected.size()-1] += 1;
+      stoneCount -= 1;
+    }
     currentWeight = getRandomWeight(totalWeight);
   }
   //if stones are left, assign junk
   if (stoneCount > 0){
-    selected[HINT_TYPE_JUNK] += stoneCount;
+    selected[selected.size()-1] += stoneCount;
   }
 }
 
-
-uint8_t PlaceHints(std::array<uint8_t, HINT_TYPE_MAX>& selectedHints,
-                std::array<HintDistributionSetting, (int)HINT_TYPE_MAX>& distTable, 
-                uint8_t* remainingDungeonWothHints,
-                uint8_t* remainingDungeonBarrenHints){
-
-  std::vector<RandomizerCheck> blankList = {};
+uint8_t PlaceHints(std::vector<uint8_t>& selectedHints,
+                std::vector<HintDistributionSetting>& distTable){
   auto ctx = Rando::Context::GetInstance();
-
-  std::array<std::vector<RandomizerCheck>, HINT_TYPE_MAX> hintTypePools = {
-      blankList, //static, should not happen
-      blankList, //trial, should not happen
-      ctx->wothLocations, //woth
-      CalculateBarrenRegions(), //Barren
-      blankList, //enternce, not yet implemented
-      blankList, // always, should not happen
-      FilterFromPool(ctx->allLocations, [](const RandomizerCheck loc){return Rando::StaticData::GetLocation(loc)->GetHint()->GetType() == HintCategory::Sometimes;}), //sometimes
-      FilterFromPool(ctx->allLocations, [](const RandomizerCheck loc){return Rando::StaticData::GetLocation(loc)->IsCategory(Category::cSong);}), //songs
-      FilterFromPool(ctx->allLocations, [](const RandomizerCheck loc){return Rando::StaticData::GetLocation(loc)->IsOverworld();}), //overworld
-      FilterFromPool(ctx->allLocations, [](const RandomizerCheck loc){return Rando::StaticData::GetLocation(loc)->IsDungeon();}), //dungeon
-      ctx->allLocations, //Named item
-      ctx->allLocations, //random
-      blankList //junk, irrelevent
-  };
-
-  for (uint8_t hintType = 0; hintType < HINT_TYPE_MAX; hintType++){
-    for (uint8_t numHint = 0; numHint < selectedHints[hintType]; numHint++){
+  uint8_t curSlot = 0;
+  for (HintDistributionSetting distribution : distTable){
+    std::vector<RandomizerCheck> hintTypePool = FilterHintability(ctx->allLocations, distribution.filter);
+    for (uint8_t numHint = 0; numHint < selectedHints[curSlot]; numHint++){
 
       SPDLOG_DEBUG("Attempting to make hint of type: ");
-      SPDLOG_DEBUG(hintTypeNames[hintType]);
+      SPDLOG_DEBUG(hintTypeNames[(int)distribution.type]);
```

Comment:
Not too concerned about it since this is just a debug print, but
```suggestion
      SPDLOG_DEBUG(hintTypeNames[static_cast<int>(distribution.type)]);
```